classes/webservice/WebserviceSpecificManagementAttachments.php

```diff
@@ -332,10 +336,14 @@ public function executeFileAddAndEdit(): void
                 unlink(_PS_DOWNLOAD_DIR_ . $attachment->file);
             }
 
+            $defaultLanguage = Configuration::get('PS_LANG_DEFAULT');
+
             $attachment->file = $file['id'];
             $attachment->file_name = $file['file_name'];
             $attachment->mime = $file['mime_type'];
-            $attachment->name[Configuration::get('PS_LANG_DEFAULT')] = $_POST['name'] ?? $file['file_name'];
+            if ($attachment->name[$defaultLanguage] === null) {
```

Comment:
```suggestion
            if (empty($attachment->name[$defaultLanguage]) || !empty($_POST['name'])) {
```

Reply:
I believe it was your suggestion that we should update the value if `$_POST['name']` is specified in the PATCH request no? And using empty is probably more adapted because an empty string is not a correct name either

Reply:
To be honest I don't remember what we said ^^ but if we use PATCH then $_POST will be empty.

From the implementation of the class it seems that PUT and PATCH request can only carry a file binary content, that is available in `php://input`, not in $_POST.

See https://github.com/PrestaShop/PrestaShop/blob/develop/classes/webservice/WebserviceSpecificManagementAttachments.php#L307

When I perform PATCH requests I can only submit a file as binary file content and there is no other body.

Reply:
Hmmm, that's not normal A patch method is supposed to allow performing partial update, so you can send only the name or only the file or both if you want
And in a PATCH method in PHP the `$_POST` variable is supposed to be filled the same way as with a POST request

Now it seems like the limitation comes from PHP which fills the `$_FILES` variable ONLY in a POST request, but there are solutions that can be found on internet, the basic workaround is to manually parse the multipart request like in this solution:
https://stackoverflow.com/questions/5483851/manually-parse-raw-multipart-form-data-data-with-php/5488449#5488449

Reply:
> Hmmm, that's not normal A patch method is supposed to allow performing partial update, so you can send only the name or only the file or both if you want

At the bottom of https://github.com/PrestaShop/PrestaShop/issues/28748 you'll find all endpoints related to this case and how they are used.

Be careful, there is a difference between endpoint `/api/attachments/` and `/api/attachments/file` 

The `/api/attachments/file` focuses only on the attached file on disk. So
- GET fetches the file content (and just the file, not its name, no description, no file extension)
- POST creates the file
- PUT modifies the file
- PATCH also modifies the file
- DELETE deletes the file

This PR aims to enable http requests against `/api/attachments/file` endpoint that contain only file, no filename additional field.